xarray/core/parallel.py

```diff
@@ -0,0 +1,86 @@
+try:
+    import dask
+    import dask.array
+    from dask.highlevelgraph import HighLevelGraph
+
+except ImportError:
+    pass
+
+from .dataarray import DataArray
+
+
+def get_chunk_slices(dataset):
+    chunk_slices = {}
+    for dim, chunks in dataset.chunks.items():
+        slices = []
+        start = 0
+        for chunk in chunks:
+            stop = start + chunk
+            slices.append(slice(start, stop))
+            start = stop
+        chunk_slices[dim] = slices
+
+    return chunk_slices
+
+
+def map_blocks(func, darray):
+    """
+    A version of dask's map_blocks for DataArrays.
```

Comment:
We should maybe not assume that the user is familiar with Dask?  I don't know how common this is among Xarray users.

```suggestion
    Apply a function across each chunk of the DataArray in parallel
    
    This is similar to Dask Array's map_blocks function, but your function 
    receives a DataArray rather than a Numpy Array.
```